microsetta_admin/tests/test_metadata_transforms.py

```diff
@@ -0,0 +1,164 @@
+                           [59.9],
+                           [60],
+                           [69.9],
+                           [70],
+                           [122],
+                           [123],
+                           [12345],
+                           [MISSING_VALUE]],
+                          columns=['age_years'],
+                          index=list('abcdefghijklmnopqrstuv'))
+        exp = pd.Series([MISSING_VALUE, 'baby', 'baby', 'child', 'child',
+                         'teen', 'teen', '20s', '20s', '30s', '30s', '40s',
+                         '40s', '50s', '50s', '60s', '60s', '70+', '70+',
+                         MISSING_VALUE, MISSING_VALUE, MISSING_VALUE],
+                        index=list('abcdefghijklmnopqrstuv'),
+                        name='age_cat')
+        self._test_transformer(AgeCat, df, exp)
+
+    def test_BMI(self):
+        df = pd.DataFrame([[180, 50],
+                           [180, 60],
+                           [170, 70],
+                           [MISSING_VALUE, 60],
+                           [150, MISSING_VALUE],
+                           [MISSING_VALUE, MISSING_VALUE]],
+                          index=list('abcdef'),
+                          columns=['height_cm', 'weight_kg'])
+        exp = pd.Series([15.4, 18.5, 24.2, MISSING_VALUE, MISSING_VALUE,
+                        MISSING_VALUE], index=list('abcdef'), name='bmi')
+        self._test_transformer(BMI, df, exp)
+
+    def test_BMICat(self):
+        df = pd.DataFrame([[-2],
+                           [7.9],
+                           [8],
+                           [18.4],
+                           [18.5],
+                           [24.9],
+                           [25],
+                           [29.9],
+                           [30],
+                           [79.9],
+                           [80],
+                           [210],
+                           [MISSING_VALUE]],
+                          index=list('abcdefghijklm'),
+                          columns=['bmi'])
+        exp = pd.Series([None, None, 'Underweight', 'Underweight', 'Normal',
+                         'Normal', 'Overweight', 'Overweight', 'Obese',
+                         'Obese', None, None, None],
+                        index=list('abcdefghijklm'), name='bmi_cat')
+        self._test_transformer(BMICat, df, exp)
+
+    def test_NormalizeWeight(self):
+        df = pd.DataFrame([[180, 'pounds'],
+                           [180, 'kilograms'],
+                           [-1, 'pounds'],
+                           [None, 'pounds'],
+                           [180, None]], index=list('abcde'),
+                          columns=['weight_kg', 'weight_units'])
+        exp = pd.Series([180 / 2.20462, 180, MISSING_VALUE, MISSING_VALUE,
+                        MISSING_VALUE],
+                        index=list('abcde'), name='weight_kg')
+        self._test_transformer(NormalizeWeight, df, exp)
```

Comment:
How does this pass with the required column name set to 'weight_cm' in the definition of NormalizeWeight above?

Reply:
The transform itself doesn't assert the required columns, which is not a good approach. will fix